pyspecdata/lmfitdata.py

```diff
@@ -266,8 +242,13 @@ def eval(self, taxis=None, set_what=None, set_to=None):
             newdata.set_error(self.fit_axis,
                     self.get_error(self.fit_axis))
         # }}}
-        newdata
-        newdata.data[:] = self.fitfunc(p, taxis).flatten()
+        param_dict = {}
+        for j in range(len(list(p))):
+            case = {list(self.parameter_names)[j]:list(p)[j]}
+            param_dict.update(case)
+        self.set_guess(param_dict)
+        self.fitfunc = self.run_lambda(self.pars)
+        newdata.data[:] = self.fitfunc.flatten()
```

Comment:
the difference in your last line here is causing a problem you noted on slack.
You should read the eval function, and see how it uses and then modifies the variable `taxis`.
You'll see that `fitfunc` and therefore `run_lambda` needs to take taxis as an argument (if `run_lambda` is used elsewhere, it might make sense to make this an optional argument).